packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -55,17 +112,17 @@ const extractObjectExpression = (node: t.ObjectExpression, meta: Metadata): CSSO
       } else if (t.isNumericLiteral(propValue)) {
         // We've found a numeric literal like: `fontSize: 12`
         value = addUnitIfNeeded(key, propValue.value);
-      } else if (t.isObjectExpression(propValue)) {
-        // We've found a nested object like: `':hover': { color: 'red' }`
-        const result = extractObjectExpression(propValue, updatedMeta);
-        css += `${key} { ${result.css} }`;
-        variables = variables.concat(result.variables);
+      } else if (t.isObjectExpression(propValue) || t.isLogicalExpression(propValue)) {
+        const result = buildCss(propValue, updatedMeta);
+        css.push(...result.css.map((x) => ({ ...x, css: `${key} { ${getItemCss(x)} }` })));
```

Comment:
nit: A comment above will be beneficial. Explains what type of scenario we are handling here. Extracting this into a function will be like cherry on top. hehe. But lets have a comment for now.